pkg/apis/node.harvesterhci.io/v1beta1/nodeconfig.go

```diff
@@ -0,0 +1,66 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConfigsApplied     = "Applied"
+	ConfigsWaitApplied = "WaitApplied"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=nodeconfigs,scope=Namespaced
+// +kubebuilder:printcolumn:name="Scope",type=string,JSONPath=`.spec.nodeName`
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.allConfigStatus`
+
+type NodeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              NodeConfigSpec   `json:"spec"`
+	Status            NodeConfigStatus `json:"status"`
+}
+
+type NodeConfigSpec struct {
+	NodeName  string     `json:"nodeName"`
+	NTPConfig *NTPConfig `json:"ntpConfigs,omitempty"`
+}
+type NTPConfig struct {
+	NTPServers     string `json:"ntpServers"`
+	ConfigModified bool   `json:"configModified"`
+}
+
+type NodeConfigStatus struct {
+	NTPConditions []ConfigStatus `json:"ntpConditions,omitempty"`
+	// AllConfigStatus present the all configs status (Applied or WaitApplied)
+	// the current state config, options are "Applied", "WaitApplied"
+	// +kubebuilder:validation:Enum:=Applied;WaitApplied
+	// +kubebuilder:default:="WaitApplied"
+	AllConfigStatus string `json:"allConfigsStatus"`
```

Comment:
I think this is a little bit redundant to the wrangler's condition types, e.g.,
```golang
type NodeConfigStatus struct {
    Conditions []Condition `json:"conditions,omitempty"`
}

type Condition struct {
	Type   condition.Cond    `json:"type"`
	Status corev1.ConditionStatus `json:"status"`
	// +nullable
	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
	// +nullable
	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
	Reason             string      `json:"reason,omitempty"`
	Message            string      `json:"message,omitempty"`
}

const NTPConfigReady condition.Cond = "NTPConfigReady"
```
so 1. we could set the condition to `False` and set a proper error message when the local node config is invalid or not consistent
2. the UI framework could leverage the existing condition design and don't need to handle the error conditions individually.

Reply:
I just simply keep the status here. We could walk all statuses or check CRD's annotation to check the applied status.